cana/boolean_network.py

```diff
@@ -164,20 +174,30 @@ def from_file(self, file, type="cnet", keep_constants=True, **kwargs):
         with open(file, "r") as infile:
             if type == "cnet":
                 return self.from_string_cnet(
-                    infile.read(), keep_constants=keep_constants, **kwargs
+                    infile.read(),
+                    keep_constants=keep_constants,
+                    partial_lut=partial_lut,
+                    **kwargs,
                 )
             elif type == "logical":
                 return self.from_string_boolean(
                     infile.read(), keep_constants=keep_constants, **kwargs
                 )
 
     @classmethod
-    def from_string_cnet(self, string, keep_constants=True, **kwargs):
+    def from_string_cnet(
```

Comment:
Move this stuff into the `boolean_node.py` file, and just call the `BooleanNode` constructor to handle clashes and unspecified outputs

Reply:
removed all boolean network changes for now. will incorporate this at a later date after all the existing work is done.